### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bohdan-mykhailenko**

### CODE REVIEW

File: src/arrayMethodSort.test.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ test(`sort strings: ['c','b','a','d'].sort2()`, () => {
 
 test(`sort cyrillic letters: ['ёж','як','аист'].sort2()`, () => {
   expect(['ёж', 'як', 'аист'].sort2())
-    .toEqual(['аист', 'як', 'ёж']);
+    .toEqual(['аист', 'ёж', 'як']);
```
Comment (dinaminahi) [inline src/arrayMethodSort.test.js:46]: You can't change the test

File: src/arrayMethodSort.js [hunk 2]
```diff
@@ -4,8 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(callback) {
+    const compareFunction = callback
+    || ((a, b) => String(a).localeCompare(String(b)));
```
Comment (dinaminahi) [inline src/arrayMethodSort.js:9]: Instead of  using || you can give this function as the default value for `callback` argument

Comment (dinaminahi) [inline src/arrayMethodSort.js:9]: The reason the test is failing is this default compare function. 
It is better to use something like this ` (a, b) => a.toString() > b.toString()`

**Review by dinaminahi — CHANGES_REQUESTED** [review]

File: src/arrayMethodSort.js [hunk 3]
```diff
@@ -3,9 +3,26 @@
 /**
  * Implement method Sort
  */
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(callback = (a, b) => String(a) > (String(b))) {
```
Comment (silchyn) [inline src/arrayMethodSort.js:8]:
```suggestion
  [].__proto__.sort2 = function(comparator = (a, b) => String(a) > String(b)) {
```

**Review by silchyn — CHANGES_REQUESTED** [review]

**Review by 33Orange — CHANGES_REQUESTED** [review]
check @silchyn comment

**Review by silchyn — APPROVED** [review]
that's cool 👍
I meant the faster bubble sort variant, but that fits

**Pull request closed without merging**